src/integration/resources/inprocess/cluster.go

```diff
@@ -54,6 +57,22 @@ type ClusterConfigs struct {
 	Aggregator *aggcfg.Configuration
 }
 
+// ClusterFullConfigs contain the final configs to use for components within
+// the cluster.
+type ClusterFullConfigs struct {
+	// DBNodes is the configuration for db nodes.
+	DBNodes []dbcfg.Configuration
+	// DBNodeOpts is the options for setting up db nodes.
+	DBNodeOpts []DBNodeOptions
```

Comment:
Do we need the DBNodeOpts and EnvConfig on here? Seems like we only need the component configs.

Reply:
DBNodeOpts is used to set up dbnodes in cluster. The seed node has a different options from others.
I can skip envconfig, then i need to get the envconfig from the coordinator config for setting up the test.

Reply:
Ahhh ok, I understand now. I think what's throwing me off here is that we have full configs mixed with options. I think these need to be broken out. What do you think about having the following structure?

```go
type ClusterSpecification struct {
  Configs PerInstanceConfigs
  ComponentOptions PerInstanceOptions
}

type PerInstanceConfigs struct {
  DBNodes []dbcfg.Configuration
  Coordinator coordinatorcfg.Configuration
  Aggregators []aggcfg.Configuration
}

type PerInstanceOptions struct {
  DBNodeOpts []DBNodeOptions

  // Note: this struct is nice because it allows us to add other component options
  // here in the future if need be.
}
```
I really like the name `ClusterSpecification` as I think it more clearly delineates that this is the exact spec of the cluster. I'm like but am less married to `PerInstanceConfigs` and `PerInstanceOptions` so open to other ideas here, but I really want us to be diligent about naming to try and make it as clear as possible as to what these structs mean to end users. I think the difference between `ClusterFullConfig` and `ClusterConfig` is a bit subtle at first glance, so trying to come up with something that really highlights the distinction.